posthog/hogql_queries/query_date_range.py

```diff
@@ -0,0 +1,110 @@
+from datetime import datetime
+from functools import cached_property
+from typing import Optional
+
+import pytz
+from dateutil.relativedelta import relativedelta
+
+from posthog.hogql.parser import parse_expr, ast
+from posthog.models.team import Team
+from posthog.queries.util import get_earliest_timestamp
+from posthog.schema import DateRange, IntervalType
+from posthog.utils import DEFAULT_DATE_FROM_DAYS, relative_date_parse, relative_date_parse_with_delta_mapping
+
+
+# Originally copied from posthog/queries/query_date_range.py with some changes to support the new format
```

Comment:
Can you clarify what are "some changes"? Should these files be kept in sync?
Can you also add copy of `test_query_date_range.py` for completion?

I don't have the full context, so hard to verify the correctness of the code and what changed just by looking at this :).